### ISSUE TITLE
Can't do anything with my unavailable pool

### ISSUE THREAD

**Issue by A2116** [openzfs/zfs#10781]
### System information

Type | Version/Name
 --- | ---

Type | Version/Name
-- | --
Distribution Name | CentOS
Distribution Version | 8.2
Linux Kernel | 5.6.15
Architecture | X86_64
ZFS Version | 0.8.4-1
SPL Version | 0.8.4-1

### Describe the problem you're observing

I have created a pool that uses partition instead of the whole disk in its log and cache devices
after failing in the log device that is occurred because of partition table become cleared the pool becomes unavailable
and showed UNAVAIL in pool and logs and it's devices
so I try to solve the problem using labelclear, clear, detach, remove commands to remove log devices and make pool work but nothing can't resolve the problem
the devices are provided to zfs by DM multipath and in all commands, it shows me the error that will come in the below

### Describe how to reproduce the problem

use multipath for the device using in pool and log
Create a pool and use partition instead of the disk for log devices
delete partitions
set /sys/module/zfs/parameters/zfs_autoimport_disable to 0
reboot

### Include any warning/errors/backtraces from the system logs

[root@Storage-Server ~]# zpool status
  pool: pool1
 state: FAULTED
   see: http://zfsonlinux.org/msg/ZFS-8000-6X
  scan: none requested
config:

        NAME                                                                                                        STATE     READ WRITE CKSUM
        pool1                                                                                                       UNAVAIL      0     0     0  missing device
          raidz3-0                                                                                                  ONLINE       0     0     0
            36000c29248dd75f04588ebd18797f59c                                                                       ONLINE       0     0     0
            36000c293a28dc7c2c613a4461c27c50f                                                                       ONLINE       0     0     0
            36000c290c6cea6770f2213c3d1eaecd7                                                                       ONLINE       0     0     0
            36000c2956e0ffbcca3c1d17967667259                                                                       ONLINE       0     0     0
            36000c2984193c5868ef7e5720424111e                                                                       ONLINE       0     0     0
          raidz3-1                                                                                                  ONLINE       0     0     0
            36000c293883ea617e8ced2d7d9709627                                                                       ONLINE       0     0     0
            36000c290d8ebd0d630421298dcab25ed                                                                       ONLINE       0     0     0
            36000c29dd8d71cdaafb99a1430e0e8ea                                                                       ONLINE       0     0     0
            36000c296fefd965583101a7d82c912a5                                                                       ONLINE       0     0     0
            36000c29ad1d5baf77874f1d7f2e2a1ff                                                                       ONLINE       0     0     0
        logs
          mirror-2                                                                                                  UNAVAIL      0     0     0  insufficient replicas
            nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000002p2  UNAVAIL      0     0     0
            nvme.15ad-564d57617265204e564d455f30303032-564d77617265205669727475616c204e564d65204469736b-00000002p2  UNAVAIL      0     0     0
          mirror-3                                                                                                  UNAVAIL      0     0     0  insufficient replicas
            nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000001p2  UNAVAIL      0     0     0
            nvme.15ad-564d57617265204e564d455f30303032-564d77617265205669727475616c204e564d65204469736b-00000001p2  UNAVAIL      0     0     0

-------------------------------------------------------------------------------

[root@Storage-Server ~]# zpool remove pool1 mirror-2 mirror-3
cannot open 'pool1': pool is unavailable

-------------------------------------------------------------------------------

[root@Storage-Server ~]# zpool clear pool1
cannot clear errors for pool1: one or more devices is currently unavailable

-------------------------------------------------------------------------------

[root@Storage-Server ~]# zpool clear pool1 nvme.15ad-564d57617265204e564d455f30303032-564d77617265205669727475616c204e564d65204469736b-00000002p2
cannot clear errors for nvme.15ad-564d57617265204e564d455f30303032-564d77617265205669727475616c204e564d65204469736b-00000002p2: one or more devices is currently unavailable

-------------------------------------------------------------------------------

[root@Storage-Server ~]# zpool labelclear -f /dev/mapper/nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000002p2
failed to open /dev/mapper/nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000002p2: No such file or directory

-------------------------------------------------------------------------------

**Comment by rincebrain** [comment]
If it were me, I'd
`zpool export pool1`
`zpool import -m pool1`

**Comment by A2116** [comment]
I tried it before, it's not exported, not imported and devices do not detach

[root@Storage-Server ~]# zpool export pool1
cannot export 'pool1': one or more devices is currently unavailable

-----------------------------------------------------------------------------------

[root@Storage-Server ~]# zpool import -m pool1
cannot import 'pool1': a pool with that name already exists
use the form 'zpool import <pool | id> <newpool>' to give it a new name
[root@Storage-Server ~]#
[root@Storage-Server ~]# zpool import -m pool1 newpool1
cannot import 'pool1': no such pool available

-----------------------------------------------------------------------------------

[root@Storage-Server ~]# zpool detach pool1 nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000002p2
cannot open 'pool1': pool is unavailable

**Comment by A2116** [comment]
I have solved this problem by disabling auto-import by removing 
options zfs zfs_autoimport_disable=0
from /etc/modprobe.d/zfs.conf file
reboot and then no pool not auto imported
then I use 
# zpool import -m pool1
then the pool status become like below

[root@Storage-Server ~]# zpool status
  pool: pool1
 state: DEGRADED
status: One or more devices could not be used because the label is missing or
        invalid.  Sufficient replicas exist for the pool to continue
        functioning in a degraded state.
action: Replace the device using 'zpool replace'.
   see: http://zfsonlinux.org/msg/ZFS-8000-4J
  scan: scrub repaired 0B in 0 days 00:00:00 with 0 errors on Mon Aug 10 02:15:52 2020
config:

        NAME                                   STATE     READ WRITE CKSUM
        pool1                                  DEGRADED     0     0     0
          raidz3-0                             ONLINE       0     0     0
            36000c29248dd75f04588ebd18797f59c  ONLINE       0     0     0
            36000c293a28dc7c2c613a4461c27c50f  ONLINE       0     0     0
            36000c290c6cea6770f2213c3d1eaecd7  ONLINE       0     0     0
            36000c2956e0ffbcca3c1d17967667259  ONLINE       0     0     0
            36000c2984193c5868ef7e5720424111e  ONLINE       0     0     0
          raidz3-1                             ONLINE       0     0     0
            36000c293883ea617e8ced2d7d9709627  ONLINE       0     0     0
            36000c290d8ebd0d630421298dcab25ed  ONLINE       0     0     0
            36000c29dd8d71cdaafb99a1430e0e8ea  ONLINE       0     0     0
            36000c296fefd965583101a7d82c912a5  ONLINE       0     0     0
            36000c29ad1d5baf77874f1d7f2e2a1ff  ONLINE       0     0     0
        logs
          mirror-2                             UNAVAIL      0     0     0  insufficient replicas
            11212847601194444581               UNAVAIL      0     0     0  was /dev/mapper/nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000002p2
            747234635590686895                 UNAVAIL      0     0     0  was /dev/mapper/nvme.15ad-564d57617265204e564d455f30303032-564d77617265205669727475616c204e564d65204469736b-00000002p2
          mirror-3                             UNAVAIL      0     0     0  insufficient replicas
            8011370189144817268                UNAVAIL      0     0     0  was /dev/mapper/nvme.15ad-564d57617265204e564d455f30303033-564d77617265205669727475616c204e564d65204469736b-00000001p2
            12418174772217149684               UNAVAIL      0     0     0  was /dev/mapper/nvme.15ad-564d57617265204e564d455f30303032-564d77617265205669727475616c204e564d65204469736b-00000001p2

errors: No known data errors

then i used
# zpool remove pool1 mirror-2
# zpool remove pool1 mirror-3

and the problem becomes solved

**Comment by rincebrain** [comment]
I'd actually suggest leaving this open, since it appears the zfs_autoimport_disable tunable led to your pool being in a problematic state you couldn't get out of using usual tools.

But that's just me. Others might have differing opinions. :)

**Comment by A2116** [comment]
I agree, especially in production environment that rebooting the system impossible or causes other problems
